todo_app/tests/test_mocking.py

```diff
@@ -0,0 +1,34 @@
+import os
+import pytest
+import mongomock
+from dotenv import load_dotenv, find_dotenv
+from todo_app import app
+from pymongo import MongoClient
+
+@pytest.fixture
+def client(monkeypatch):
+    file_path = find_dotenv(".env.test")
+    load_dotenv(file_path, override=True)
+
+    mock_client = mongomock.MongoClient()
+    monkeypatch.setattr(MongoClient, "__init__", lambda self, *args, **kwargs: None)
+    monkeypatch.setattr(MongoClient, "get_database", lambda self, *args, **kwargs: mock_client['todo_app_db'])
+
+    test_app = app.create_app()
+    with test_app.test_client() as client:
+        yield client
```

Comment:
This seems very reasonable, _but_ I'm not convinced it's quite working as you want - in particular the circumstances in which this works for me correlate with the circumstances where real env variables are being loaded at least part of the time. For example, as this currently stands, building the docker test image, and running it with only the fake variables (`docker run --env-file .env.test <image_name> todo_app/tests`) errors for me, even if we deal with the `.env.test` quotes mentioned above.

For this reason, I'd suggest switching to the pattern of patching mentioned:
```suggestion
    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
        test_app = app.create_app()
        with test_app.test_client() as client:
            yield client
```
Which **does** also require that we update the pymongo import to be `import pymongo` rather than `from pymongo import MongoClient`, and therefore update line 23 to: `db = pymongo.MongoClient().get_database("todo_app_db")`